src/upchunk.ts

```diff
@@ -314,118 +494,154 @@ export class UpChunk  {
       headers,
       url: this.endpointValue,
       method: this.method,
-      body: this.chunk,
+      body: chunk,
     });
   }
 
-  /**
-   * Called on net failure. If retry counter !== 0, retry after delayBeforeAttempt
-   */
-  private manageRetries() {
-    if (this.attemptCount < this.attempts) {
-      setTimeout(() => this.sendChunks(), this.delayBeforeAttempt * 1000);
+  protected async sendChunkWithRetries(chunk: Blob): Promise<boolean> {
+    // What to do if a chunk was successfully uploaded
+    const successfulChunkUploadCb = async (res: XhrResponse, _chunk?: Blob) => {
+      // Side effects
+      const lastChunkEnd = new Date();
+      const lastChunkInterval =
+        (lastChunkEnd.getTime() - this.lastChunkStart.getTime()) / 1000;
```

Comment:
if using `Date.now()` update here too. it also saves some characters

Reply:
same as above